receiver/haproxyreceiver/factory.go

```diff
@@ -0,0 +1,81 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package haproxyreceiver // import "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/haproxyreceiver"
+
+import (
+	"context"
+	"time"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/config"
+	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/receiver"
+	"go.opentelemetry.io/collector/receiver/scraperhelper"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/haproxyreceiver/internal/metadata"
+)
+
+const (
+	typeStr   = "haproxy"
+	stability = component.StabilityLevelDevelopment
+)
+
+// NewFactory creates a new HAProxy receiver factory.
+func NewFactory() receiver.Factory {
+	return receiver.NewFactory(
+		typeStr,
+		CreateDefaultConfig,
+		receiver.WithMetrics(createReceiverFunc(), stability))
+}
+
+func CreateDefaultConfig() component.Config {
+	return &Config{
+		ReceiverSettings: config.NewReceiverSettings(component.NewID(typeStr)),
+		ScraperControllerSettings: scraperhelper.ScraperControllerSettings{
+			ReceiverSettings:   config.NewReceiverSettings(component.NewID(typeStr)),
+			CollectionInterval: 1 * time.Minute,
+		},
+		MetricsSettings: metadata.DefaultMetricsSettings(),
+	}
+}
+
+func createReceiverFunc() receiver.CreateMetricsFunc {
```

Comment:
Same again with the naming convention:

```suggestion
func newReceiver() receiver.CreateMetricsFunc {
```